community/kernel-api/src/main/java/org/neo4j/internal/kernel/api/Transaction.java

```diff
@@ -61,7 +63,7 @@
     /**
      * @return The Write operations of the graph.
      */
-    Write dataWrite();
+    Write dataWrite() throws KernelException;
```

Comment:
I would add it to the javadoc that the KernelException is thrown for instance in the case where the transaction cannot be upgraded to a write transaction, e.g. when it has already done schema modifications.